protocols/relay/src/priv_client/handler.rs

```diff
@@ -458,38 +351,167 @@ impl ConnectionHandler for Handler {
             return Poll::Ready(ConnectionHandlerEvent::Close(err));
         }
 
+        // Circuit connections
+        if let Poll::Ready(worker_res) = self.circuit_connection_futs.poll_unpin(cx) {
+            let res = match worker_res {
+                Ok(r) => r,
+                Err(Timeout { .. }) => {
+                    return Poll::Ready(ConnectionHandlerEvent::Close(StreamUpgradeError::Timeout));
+                }
+            };
+
+            let opt = match res {
```

Comment:
Instead of assigning to `opt` here, I think I'd prefer just using `return` within the branches were we need to.